### PR TITLE
8253736: Cleanup some of WorkArounds and usage thereof

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
This cleanup is primarily focussed on the WorkArounds class, simplifying/removing code that is no longer required.

Also, simplified DocLint initialization such that it no longer needs a `JavacTask` and so the initialization code can be moved out of `WorkArounds`.

The "constant value expression" in Utils is simplified to remove a redundant level of enclosing nested class for the type-kind visitor.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253736](https://bugs.openjdk.java.net/browse/JDK-8253736): Cleanup some of WorkArounds and usage thereof

### Reviewers
 * [Vicente Romero](https://openjdk.java.net/census#vromero) (@vicente-romero-oracle - **Reviewer**)
 * [Kumar Srinivasan](https://openjdk.java.net/census#ksrini) (@kusrinivasan - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/412/head:pull/412`
`$ git checkout pull/412`

### CODE REVIEW

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/WorkArounds.java [hunk 1]
```diff
@@ -94,110 +95,36 @@
...
-        return (doclint == null);
+        // Note: this one use of DocEnvImpl is what prevents us tunnelling extra
+        // info from a doclet to its taglets via a doclet-specific subtype of
+        // DocletEnvironment.
+        toolEnv = ((DocEnvImpl)this.configuration.docEnv).toolEnv;
+        javacTypes = toolEnv.getTypes();
     }
 
     /*
      * TODO: This method exists because of a bug in javac which does not
```
Comment (vicente-romero-oracle) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/WorkArounds.java:117]: probably a good idea to include the bug ID here?

**Review by vicente-romero-oracle — APPROVED** [review]
just a nit comment, no need for another iteration, the changes looks good to me

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/WorkArounds.java (hunk 1 above)
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/WorkArounds.java:117]: @vicente-romero-oracle This relates to a discussion we should have elsewhere, separately, about the interactions between deprecation and packages, and whether we should support depicted packages. JLS does not list packages in the list of kinds of elements that can be deprecated, in which case this comment is probably incorrect, as the bug may actually be here in javadoc for trying to support deprecated packages!

**Comment by jonathan-gibbons** [comment]
@vicente-romero-oracle thanks for the review.
I raised the number of reviewers to have a javadoc reviewer participate as well.

**Review by kusrinivasan — APPROVED** [review]
Nice to see WorkArounds being chipped away. It is not clear to me if additional tests are warranted for getConstantValue for TestConstantValuesDriver.java, in langtools test I saw only ElementStructureTest.java. I will leave it to you, maybe file a follow up issue.

File: test/langtools/jdk/javadoc/doclet/constantValues/TestConstantValuesDriver.java [hunk 2]
```diff
@@ -57,15 +57,15 @@ public void test() {
                 """
                     <code id="TestConstantValues.BYTE_MAX_VALUE">public&nbsp;static&nbsp;final&nbsp;byte</code></td>
                     <th class="col-second" scope="row"><code><a href="TestConstantValues.html#BYTE_MAX_VALUE">BYTE_MAX_VALUE</a></code></th>
-                    <td class="col-last"><code>127</code></td>""",
+                    <td class="col-last"><code>0x7f</code></td>""",
```
[the remark this answers is not in the document]
Comment (jonathan-gibbons) [reply test/langtools/jdk/javadoc/doclet/constantValues/TestConstantValuesDriver.java:60]: https://bugs.openjdk.java.net/browse/JDK-8253941

**Pull request closed without merging**